### PR TITLE
Upgrade to Ruby 3

### PR DESCRIPTION
**Pull request by mitchellhenke**
Ruby 2.7 is only receiving security updates so we should upgrade to Ruby 3.0

### CODE REVIEW

File: .ruby-version [hunk 1]
```diff
@@ -1 +1 @@
-2.7
+3.0
```
Comment (aduth) [inline .ruby-version:1]: Could we go to 3.1?

**Comment by aduth** [comment]
I'm seeing this error when testing locally after being redirected back to the application. Is it related?

>REXML::ParseException at /consume
>Garbage component exists at the end: <]>: </p:Response[ID=$id]/ds:Signature]>
>file: xpathparser.rb location: parse line: 28

File: .ruby-version (hunk 1 above)
Comment (mitchellhenke) [reply .ruby-version:1]: I wish, but the infrastructure pieces don't have it installed just yet (but I have a [PR](https://github.com/18F/identity-base-image/pull/230))

**Comment by mitchellhenke** [comment]
@aduth great catch, we also need to update `ruby-saml`

**Review by mdiarra3 — APPROVED** [review]

**Pull request merged**